src/database.ts

```diff
@@ -290,3 +291,14 @@ export class DatabaseTable<Row> {
 		return this.updateAll(data)`WHERE \`${this.primaryKeyName}\` = ${primaryKey} LIMIT 1`;
 	}
 }
+
+export class PGDatabase<T = any> {
+	database: pg.Pool | null;
+	constructor(config: pg.PoolConfig | null) {
+		this.database = config ? new pg.Pool(config) : null;
+	}
+	query(query: string, values: BasicSQLValue[]) {
+		if (!this.database) return null;
+		return this.database?.query(query, values).then(x => x.rows) as Promise<T[]>;
```

Comment:
Something's wrong if the if statement isn't filtering out the null condition, requiring you to use `?` here.

Also, maybe just an async function here, I don't know if you're writing this because you want to avoid a Promise if there's no database but it gives the function a complicated type & async/await is faster than using Promise methods like `.then`

Reply:
Yeah, made it async instead. For the former, that was just a typo.